pip_audit/format/cyclonedx.py

```diff
@@ -0,0 +1,47 @@
+from typing import Dict, List
+
+from cyclonedx import output  # type: ignore
+from cyclonedx.model.bom import Bom  # type: ignore
+from cyclonedx.model.component import Component  # type: ignore
+from cyclonedx.model.vulnerability import Vulnerability  # type: ignore
+from cyclonedx.parser import BaseParser  # type: ignore
+
+import pip_audit.service as service
+
+from .interface import VulnerabilityFormat
+
+
+class PipAuditResultParser(BaseParser):
+    def __init__(self, result: Dict[service.Dependency, List[service.VulnerabilityResult]]):
+        super().__init__()
+
+        for (dep, vulns) in result.items():
+            c = Component(name=dep.name, version=str(dep.version))
+
+            # TODO(ww): Figure out if/how we want to include other dependency
+            # metadata in the BOM, such as author, license, etc.
+
+            for vuln in vulns:
+                # TODO(ww): Figure out if/how we want to include other vulnerability
+                # metadata in the BOM, such as source (OSV/PyPI/etc), URL, etc.
+                c.add_vulnerability(
+                    Vulnerability(
+                        id=vuln.id,
+                        description=vuln.description,
+                        advisories=[f"Upgrade: {v}" for v in vuln.fix_versions],
```

Comment:
N.B.: The CycloneDX `Vulnerability` model doesn't have a field for upgrade advice, so I put it in a set of advisories instead.